router/worker.go

```diff
@@ -985,14 +985,18 @@ func (w *worker) postStatusOnResponseQ(respStatusCode int, payload json.RawMessa
 		}
 	}
 
-	inputPayload := payload
-	switch errorAt {
-	case routerutils.ERROR_AT_TF:
-		inputPayload = destinationJobMetadata.JobT.EventPayload
-		status.ErrorResponse = misc.UpdateJSONWithNewKeyVal(status.ErrorResponse, "failureStage", "RudderStack Transformation Error")
-	default: // includes ERROR_AT_DEL, ERROR_AT_CUST
-		status.ErrorResponse = misc.UpdateJSONWithNewKeyVal(status.ErrorResponse, "failureStage", "Destination Error")
+	inputPayload := destinationJobMetadata.JobT.EventPayload
+	if !w.rt.reportJobsdbPayload.Load() { // TODO: update default/remove this flag after monitoring the payload sizes
+		inputPayload = payload
+	}
+	tags := stats.Tags{
+		"destType":      w.rt.destType,
+		"workspaceID":   destinationJobMetadata.WorkspaceID,
+		"state":         status.JobState,
+		"destinationID": destinationJobMetadata.DestinationID,
 	}
+	stats.Default.NewTaggedStat("router_transformation_payload_size_bytes", stats.HistogramType, tags).Observe(float64(len(destinationJobMetadata.JobT.EventPayload)))
```

Comment:
Why do we need `router_transformation_payload_size_bytes`? Also it's router input payload rt. If we're planning to capture, may be capture at early stage?

Reply:
Also, we don't need to capture this multiple times for each router job. We can capture this once per destinationJob.